File: back/src/main/java/com/backend/restservice/models/Roster.java
package com.backend.restservice.models;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity // This tells Hibernate to make a table out of this class
public class Roster {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer rosterID;
    private Integer campusID;
    private String teacherName;
    private String teacherLastName1;
    private String teacherLastName2;
    private String subjectName;

    public Integer getRosterID() {
        return rosterID;
    }

    public Integer getCampusID() {
        return campusID;
    }

    public void setCampusID(Integer campusID) {
        this.campusID = campusID;
    }

    public String getSubjectName() {
        return subjectName;
    }

    public void setSubjectName(String subjectName) {
        this.subjectName = subjectName;
    }

    public String getTeacherLastName2() {
        return teacherLastName2;
    }

    public void setTeacherLastName2(String teacherLastName2) {
        this.teacherLastName2 = teacherLastName2;
    }

    public String getTeacherLastName1() {
        return teacherLastName1;
    }

    public void setTeacherLastName1(String teacherLastName1) {
        this.teacherLastName1 = teacherLastName1;
    }

    public String getTeacherName() {
        return teacherName;
    }

    public void setTeacherName(String teacherName) {
        this.teacherName = teacherName;
    }


}

File: front/_models/roster.ts
export default class Roster {
    roster_id?: number;
    teacher_name: string;
    teacher_lastname1: string;
    teacher_lastname2: string;
    subject_name: string;
    uni_structure_id: number;
    structure_name: string;
    structure_type: string;
    campus_id: number;
    campus_name: string;
    /**
     *
     */
    constructor(roster_id: number = 0, teacher_name: string, teacher_lastname1: string, teacher_lastname2: string,
        subject_name: string, uni_structure_id: number, structure_name: string, structure_type: string, campus_id: number,
        campus_name: string) {
        this.roster_id = roster_id;
        this.teacher_name = teacher_name;
        this.teacher_lastname1 = teacher_lastname1;
        this.teacher_lastname2 = teacher_lastname2;
        this.subject_name = subject_name;
        this.uni_structure_id = uni_structure_id;
        this.structure_name = structure_name;
        this.structure_type = structure_type;
        this.campus_id = campus_id;
        this.campus_name = campus_name;
    }
}